drivers/sensor/mcp9808/mcp9808.c

```diff
@@ -35,6 +35,39 @@ int mcp9808_reg_read(const struct device *dev, uint8_t reg, uint16_t *val)
 	return rc;
 }
 
+int mcp9808_reg_write_16bit(const struct device *dev, uint8_t reg,
+			    uint16_t val)
+{
+	const struct mcp9808_data *data = dev->data;
+	const struct mcp9808_config *cfg = dev->config;
+	uint8_t buf[3] = {
+		reg,
+		val >> 8,       /* big-endian register storage */
+		val & 0xFF,
+	};
+
+	return i2c_write(data->i2c_master, buf, sizeof(buf), cfg->i2c_addr);
+}
+
+int mcp9808_reg_write_8bit(const struct device *dev, uint8_t reg,
+			   uint8_t val)
+{
+	const struct mcp9808_data *data = dev->data;
+	const struct mcp9808_config *cfg = dev->config;
+	uint8_t buf[3] = {
+		reg,
+		val,
+	};
+
+	return i2c_write(data->i2c_master, buf, sizeof(buf), cfg->i2c_addr);
+}
+
+static int mcp9808_set_temperature_resolution(const struct device *dev,
+					      uint8_t resolution)
```

Comment:
I'm wondering if this is the correct indentation. I think 'uint8_t' should be aligned under 'const'. But I also see that this would break the other indentations in the driver. I don't know what to suggest, if leaving as it is in accordance with the rest of the driver, or to fix it.

Reply:
Happy with either convention. Small personal preference to align `uint8_t` with `const`. That is how it's showing on my device, but I need to change the indentation correctly. I'll update this and re-push.